### ISSUE TITLE
Avoid recompilations in mask attention

### ISSUE THREAD

**Issue by samvanstroud** [samvanstroud/hepattn#57]
It looks like `attn_mask[torch.where(attn_mask.sum(-1) == 0)] = True` is causing issues: 

```
site-packages/torch/_dynamo/convert_frame.py:964] [11/8] torch._dynamo hit config.recompile_limit (8)
W0701 09:46:42.634000 65418 site-packages/torch/_dynamo/convert_frame.py:964] [11/8]    function: 'torch_dynamo_resume_in_forward_at_50' (/share/rcifdata/svanstroud/hepattn/src/hepattn/models/decoder.py:50)
W0701 09:46:42.634000 65418 site-packages/torch/_dynamo/convert_frame.py:964] [11/8]    last reason: 11/5: 2 <= ___stack2[0].size()[0] <= 2147483647  # attn_mask[torch.where(attn_mask.sum(-1) == 0)] = True  # src/hepattn/models/decoder.py:50 in torch_dynamo_resume_in_forward_at_50 (user code shown is first use of this value--the guard itself is not due user code but due to 0/1 specialization in the framework; to avoid specialization try torch._dynamo.mark_unbacked(tensor, dim)) and (_inductor/codecache.py:1080 in _lookup_graph)
W0701 09:46:42.634000 65418 site-packages/torch/_dynamo/convert_frame.py:964] [11/8] To log all recompilation reasons, use TORCH_LOGS="recompiles".
W0701 09:46:42.634000 65418 site-packages/torch/_dynamo/convert_frame.py:964] [11/8] To diagnose recompilation issues, see https://pytorch.org/docs/main/torch.compiler_troubleshooting.html.
```
